src/python/pants/backend/scala/dependency_inference/scala_parser_test.py

```diff
@@ -411,6 +412,28 @@ def test_package_object(rule_runner: RuleRunner) -> None:
     assert sorted(analysis.provided_symbols) == ["foo.bar", "foo.bar.Hello"]
 
 
+def test_source3(rule_runner: RuleRunner) -> None:
+    rule_runner.set_options(
+        args=[
+            "-ldebug",
+            "--scala-version-for-resolve={'jvm-default':'2.13.8'}",
+            "--scalac-args=['-Xsource:3']",
+        ],
+        env_inherit=PYTHON_BOOTSTRAP_ENV,
+    )
+    analysis = _analyze(
+        rule_runner,
+        textwrap.dedent(
+            """
+            package foo
+
+            import bar.*
+            """
+        ),
+    )
+    assert set(analysis.all_imports()) == {"bar"} # Without source3 the import would be "bar.*"
```

Comment:
Maybe worth just directly testing that the `is_wildcard` flag was set? You can just do a comparison on `imports_by_scope` directly.

Reply:
done.